### ISSUE TITLE
shutdown: Timer too close

### ISSUE THREAD

**Issue by jimthouston** [KevinOConnor/klipper#594]
Hi Kevin,

I'm very impressed by klipper. 

I'm playing with a 3D printer which I bought  at a hamfest for parts.  It is based on a printrboard controller, but I started out using a Mega2560+RAMPS controller, and I was lucky to get a good print on my first try.  

Now I have switched back to the printrboard Rev F6 controller, and I'm getting the "Timer too close" error. The failure happens very quickly with the default max_velocity and max_accel values.  I tried cranking these values down to max_velocity: 10 and max_accel: 100.  This ran for almost 4 hours before failing with the same error.  It almost finished a 2cm cube, so slowing things down doesn't solve the problem.

The log file is from a quick failure with max_velocity:300.

I wondered if I had a divided clock so I check the fuses setting.  I think these are ok?
Fuses OK (E:F6, H:DB, L:DE)

I'm enjoying chasing this but welcome suggestions on how to debug this.
Thanks Jim
[klippy.log](https://github.com/KevinOConnor/klipper/files/2337329/klippy.log)

**Comment by jimthouston** [comment]
Thank you.  This make sense.  My problem has also changed to a hard failure with a slight hint of dead MCU smoke.  This confirms problems with power supply or grounding.  The last command that worked was turning off the motors.

When I get this patched up I will watch the bytes_retransmit value in the log.

**Comment by jimthouston** [comment]
I'm still chasing this problem.  I have done the obvious things trying different USB cables and different power supplies for the Raspberry Pi and the controller.  I have two printrboard controllers and both fail
the same way.  The only part I have not swapped out yet is the Raspberry Pi.  I guess it is next.

I liked the electrical noise idea.  The re-transmits don't seem to happen when it isn't printing.  It would make sense that noise from driving the stepper motors could couple into the power to the MCU.  Before I going wild with bypass caps I decided to do a test running the printer with the stepper motors disabled.  I changed the enable_pin for each motor to an unused pin and added a [homing_override] so that it would bypass the end stop checks.  This still failed within a few minutes.  So there still could be noise from the extruder heater switching but I'm a little less convinced that the re-transmits are just a hardware problem.
 
I'm wondering if anyone is having better luck with a printrboard?

I'm thinking about digging into the usb_serial code.
